dds/DCPS/DataReaderImpl.cpp

```diff
@@ -572,23 +572,19 @@ DataReaderImpl::remove_associations(const WriterIdSeq& writers,
 }
 
 void
-DataReaderImpl::remove_or_reschedule(const PublicationId& pub_id)
+DataReaderImpl::remove_publication(const PublicationId& pub_id)
 {
-  ACE_WRITE_GUARD(ACE_RW_Thread_Mutex, write_guard, this->writers_lock_);
-  WriterMapType::iterator where = writers_.find(pub_id);
-  if (writers_.end() != where) {
-    WriterInfo& info = *where->second;
-    WriterIdSeq writers;
-    push_back(writers, pub_id);
-    bool notify = info.notify_lost_;
-    if (info.removal_deadline_ < ACE_OS::gettimeofday()) {
-      write_guard.release();
-      remove_associations_i(writers, notify);
-    } else {
-      write_guard.release();
-      remove_associations(writers, notify);
+  WriterIdSeq writers;
+  bool notify = false;
+  {
+    ACE_WRITE_GUARD(ACE_RW_Thread_Mutex, write_guard, this->writers_lock_);
+    WriterMapType::iterator where = writers_.find(pub_id);
+    if (writers_.end() != where) {
```

Comment:
This doesn't preserve the semantics of the original code in the case where the find() was not successful.  In order to avoid further complicating the other parts of the function, if `where` is equal to the end() iterator, go ahead and return at the time it's detected.